[Update] 게시판 댓글 구현 및 세부 사항 업데이트

GisuHwang opened the pull request:
불필요 이미지 삭제한것 까지 PR 쏩니다.
레퍼런스 연결 부분이 생각처럼 안되서 일단은 형님이 말하셨던 BoardDev에 넣지 못했습니다...
다음 PR때는 어떻게든 넣어볼게요.

GisuHwang committed:
[Update] 게시판 파이어베이스 구조변경 및 게시글 작성 로직 수정

기존에 데이터를 다시 조회하여 카운트 갯수를 측정하던부분에 오류가있었습니다. 만약 삭제시 갯수카운트를 체크하지 못햇던 부분이 있어 구조를 명언 좋아요 와 비슷한 구조로 변경했습니다. 이제 게시글 삭제와 댓글 마무리 작업 진행할게요. 그리고 세부적인 작업들도 진행하겠습니다. 작업하는 동안은 아마 리스트가 나오지 않을거같아요. !

GisuHwang committed:
[Update] 게시판 댓글 구현 및 세부 사항 업데이트

게시판 리스트 스와이프 삭제 구현, 게시판 리스트에 댓글수 구현, 댓글 좋아요 구현 및 시간순 정렬(최신글 최상위) 그리고 게시글 작성시 이미지 있는 경우 트랜젝션 관리 가 되지 않아 힘든점이 있네요. 자세히 말씀드리자면 스토리지 저장하는 타이밍과 runTranSactionBlock을 통해 글이 등록이 되는데 이때 클로져이다 보니 아무래도 스토리지 저장 시에 동시에 다수사용자가 등록시 게시글 번호가 중복되는 문제가 있엇습니다. 그래서 이미지가 있을경우 스토리지에 먼저 저장후 글이 파베에 저장되도록 구조 변경하였습니다. 이렇게 함으로써 사용자글 중복은 막았고, 글쓰기 화면에서 2~3초 딜레이가 있어 인디케이터 구동했습니다.

GisuHwang committed:
[Update] 게시판 댓글 구현 및 세부 사항 업데이트

코드리뷰로 말씀하셨던 부분 수정하여 다시 커밋합니다.
Auth사용하여 currentuser uid 로 변경하였으며, 메인쓰레드 부분은 카톡으로 말씀드렸듯이 xcode내에서 빌드시 발생하는 부분이라 메인쓰레드 사용하였고, 불필요 주석들 및 잘못된 주석 변경 수정하였습니다. 추가적으로 게시글 쓰기, 댓글 쓰기시 기본적인 벨리데이션 추가하였습니다.

GisuHwang committed:
[Update] 불필요 이미지 삭제

이미지 삭제 처리 했습니다.

blackturtle2 commented on QuoteDev/QuoteDev/BoardDevListViewController.swift:
```diff
@@ -60,123 +60,67 @@ class BoardDevListViewController: UIViewController {
-        // My top posts by number of stars
-        let myTopPostsQuery = reference.child("board").queryOrdered(byChild: "board_count")
-        print("쿼리://",myTopPostsQuery)
         myTopPostsQuery.observe(.value, with: { (data) in
```
~이전에 남겼는지 기억이..~ single observe를 권장하고, observe를 사용할 이유가 있다면, `리스너 분리` 과정을 작성해야 할 것 같습니다.

https://firebase.google.com/docs/database/ios/read-and-write?hl=ko
위 링크에서 `리스너 분리` 부분 참고하셔요.

GisuHwang replied:
넵넵 저번 PR에거 싱글로 사용하라고 하셧어요
제가 댓글이나 게시글이 동시에 시뮬로 여러 기기 테스트 하다보니 각각 동기화가 되어 이게 맞는거 같아 욕심부린거 같네요..생각해보면 굳이 실시간으로 안해도 게시글을 남기거나 댓글을 보면 호출되는거네여...

blackturtle2 commented on QuoteDev/QuoteDev/BoardDevListViewController.swift:
```diff
@@ -193,22 +137,34 @@ extension BoardDevListViewController: UITableViewDelegate, UITableViewDataSource
         })
-    
         
+        self.reference.child("board_comment").child(self.boardArrs[indexPath.row].board_uid).observe(.value, with: { (dataSnap) in
```
여기도 observe만 사용되었네요.
댓글을 가져오는 부분인데, 왜 observe인지 궁금합니다.

blackturtle2 commented on QuoteDev/QuoteDev.xcworkspace/xcshareddata/WorkspaceSettings.xcsettings:
```diff
@@ -0,0 +1,5 @@
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict/>
+</plist>
```
혹시 얘는 왜 추가된 건지 알 수 있을까요? 별건 아닌데.. 왜 생겼을까. 하는..

GisuHwang replied:
제가 파일경로를 변경하다보니 커밋전에 확인할때 추가되어잇더라구요....

blackturtle2 reviewed (commented):
1. 그냥.. finder에서 swift 파일들을 원하는 폴더로 옮기고, 옮긴 파일들은 Xcode에서 빨간색으로 표시될텐데, 지우고 다시 Xcode에 옮긴 파일들을 드래그하면 됩니다.
아니면, Xcode에서 파일들의 Reference 위치(폴더 위치)를 수정하면 되는데, 파일들이 많아서 귀찮을 거여요.

2. observe로 된 부분들이 있는데, `single observe`를 안 쓰고, observe로 했는지 이유 알려주세요.
그리고 observe 사용하려면, `리스너 분리` 작업을 꼭 삽입해주세요.
리스너 분리는 앱 전체에 영향 가는 부분입니당.

 - 참고 링크 1: https://firebase.google.com/docs/database/ios/read-and-write?hl=ko#detach_listeners
 - 참고 링크 2: https://github.com/LunarFlash/Chaty/blob/master/ChatChat/ChatViewController.swift#L59

3. 파일 삭제한건 확인했는데, `delete_icon.png`는 왜 안지웠나요??

---

// KingFisher가 필요하면, podfile에 써서, pod install 해놓고, 작업 싹 해놓은 다음, Git에 Push할 때, podfile만 다시 원상 복구해놓고, push하는 방법이 있습니다.
// 혹은.. 사실 가장 간단한 방법은 내 브랜치에 있는 Podfile을 가져다가 복붙해놓으면, 나중에 충돌나지 않아요.


위 3가지만 해결되면, `Approve`하겠습니다.

GisuHwang commented:
1.레퍼런스를 위에 방법으로 햇엇는데 ?로 나오고 빌드가 실패해서 제가 잘못한거같아 다시 원상태로 PR햇네요😅
다시 확인해서 다시 할게요🙏

2.위에 리플달긴햇지만 제가 생각을 잘못한거같아요 글이 추가될때마다 다른기기도 변화는게 맞다고 착각을... 수정할게여

3. 형 브랜치에서 pod file복붙에서 작업할게요.

🙏

GisuHwang committed:
[Update] singleObserve로 변경 및 파일 구조 변경

기존 observe로 사용하던 부분을 singleObserve로 변경 하였고 PodFile에 Kingfisher 추가 하여 코드 변경 및 파일 구조 변경하였습니다.
